### DISCUSSION TITLE
Comment créer et ajouter un sprite ?

### DISCUSSION THREAD

**Discussion by Fosowl** [TempoDev/Arcade-Toulouse#53] [Questions]
Bonjour,

Est-it possible d'avoir une explication / un example sur comment créer un sprite ?

Je suppose qu'au sein de notre jeu il est nécessaire de pouvoir charger les sprites qui composent nos entitées à l'aide d'un path, comme par exemple `"assets/pacman.bmp"` 

Or je m'attendais à voir une méthode dans la structure `Sprite` ou dans la classe `IComponent` dans le style `Sprite(std::string path)` qui chargerai le fichier `assets/pacman.png` dans le `std::vector<Color> pixels` mis à notre disposition.

Merci de bien m'expliquer comment faire avec si possible un exemple ;)

**Comment by nasso** [comment]
Le format BMP est assez facile a lire toi même à la main!
Tu peux également sûrement utiliser une librairie qui le ferait pour toi (à voir avec la pedago en amont je pense).

Dans tout les cas, il aurait fallut lire et décoder l'image à un moment ou a un autre, que ça soit dans le core ou dans le jeu. On à décidé de faire ça du côté jeu pour simplifier le core et laisser le jeu choisir du format dans lequel ses ressources sont stockées.

Tu pourrais avoir un truc comme ça:

```cpp
Sprite load_bitmap(const std::string& path);

try {
    Sprite spr = load_bitmap("assets/pacman.bmp");
} catch (const AssetIOException& e) {
    std::cerr << "Couldn't load pacman sprite: " << e.what() << std::endl;
}
```

À toi de coder `load_bitmap` ;)
(c'est pas si compliqué si tu reste sur du format BMP)

**Reply by nasso** [reply]
Tu pourrais également faire un format spécialisé et lire ton propre format d'image si ça t'arrange.

Par exemple, tu pourrais faire un script Python ou autre qui convertis n'importe quelle image en format binaire convenable pour tes jeux pour te faciliter la tâche en C++. Ce format pourrait par exemple contenir uniquement la largeur, la hauteur puis les pixels dans un format connu (par exemple RGBA). L'avantage c'est que tu serais libre d'utiliser n'importe quelle librairie de chargement d'image depuis ton script de conversion!

**Reply by Fosowl** [reply]
ok merci beaucoup, c'est plus clair je penser effectivemment que le chargement des images s'effectuer au niveau du core :)

Sinon je vais éviter de me compliquer trop la vie avec un script python je pense dèja etre en négatif niveau temps :rofl:

**Reply by nasso** [reply]
Justement je proposais cette idée parce qu'elle peut potentiellement être plus rapide si tu galère trop à lire le fichier BMP :p
